src/queue/queue.ts

```diff
@@ -0,0 +1,61 @@
+import Nodee from '../list/node';
+import Storage from '../utils/storage';
+
+export default class Queue<T> {
+  private memory = new Storage<Nodee<T>>();
+
+  private firstPosition: number | null = null;
+
+  private lastPosition : number | null = null;
+
+  public length = 0;
+
+  enQueue(element: T):boolean {
+    if (!this.firstPosition) {
+      const newNode = new Nodee(element);
+      const newIndexNode = this.memory.save(newNode);
+      this.firstPosition = newIndexNode;
+      this.lastPosition = newIndexNode;
+      this.length += 1;
+      return true;
+    }
+    if (!this.lastPosition) {
+      return false;
+    }
+    const newNode = new Nodee(element);
```

Comment:
this is the same you have in the first if, so once you move the other if to the top, put the line 15 and 16 to be used always